src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -115,6 +116,22 @@ internal IEnumerable<IKeyValueAdapter> Adapters
         /// </summary>
         internal bool IsOfflineCacheConfigured { get; private set; } = false;
 
+        /// <summary>
+        /// The <see cref="ILoggerFactory"/> for creating a logger to log errors during refresh operation.
+        /// </summary>
+        internal ILoggerFactory LoggerFactory { get; set; }
+
+        /// <summary>
+        /// Register a logger factory to log errors during refresh operations.
+        /// </summary>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> for creating a logger.</param>
+        /// <returns></returns>
+        public AzureAppConfigurationOptions RegisterLoggerFactory(ILoggerFactory loggerFactory)
```

Comment:
`RegisterLoggerFactory`. Where does the naming come from? I know we have existing set-like APIs with `SetSecretResolver`.

Reply:
I didn't use `SetLoggerFactory` because we discussed that that might be confusing as `Set<property>` is the setter format in Java. But I do like it more than `RegisterLoggerFactory` because we have other `Set` APIs like your pointed out.

Reply:
Yeah, I think `SetSecretResolver` has that same problem, but now that I'm seeing it again, it looks like consistency is the way to go.

Reply:
Updated